src/v/cluster/metadata_cache.cc

```diff
@@ -111,9 +111,9 @@ ss::future<std::vector<broker_ptr>> metadata_cache::all_alive_brokers() const {
 
     std::set<model::node_id> brokers_with_health;
     for (auto& st : res.value()) {
+        brokers_with_health.insert(st.id);
         if (st.is_alive) {
             auto broker = _members_table.local().get_broker(st.id);
-            brokers_with_health.insert(st.id);
```

Comment:
ii don't follow, this seems to do the opposite. this says

whether or not is alive, add it. Don't you want it to say, *only, if it is alive*?

Reply:
based on the commit, i expected to see something lik e

`if ( some_list.empty() ) { return {}; }` kind of thing based on the commit.

Reply:
The idea behind `brokers_with_health` is that it should represent those brokers which status is already registered in the health monitor.